components/procore_sandbox/common/utils.mjs

```
@@ -0,0 +1,55 @@
+
+function buildPropDefinitions({
+  app = {}, props = {},
+}) {
+  return Object.entries(props)
+    .reduce((newProps, [
+      key,
+      prop,
+    ]) => {
+      if (!prop.propDefinition) {
+        return {
+          ...newProps,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)